src/hooks/useWithdrawLiquidityHandler.tsx

```diff
@@ -22,342 +18,201 @@ export const useWithdrawLiquidityHandler = () => {
+        setEnableApprove(true)
+      } else {
+        setEnableApprove(false)
+      }
       setsliderPercentage(value)
     }
   }
 
-  async function approveTransaction() {
-    // Position Manager contract address
-    let address =
-      process.env.NEXT_PUBLIC_ENVIRONMENT === 'local'
-        ? process.env.NEXT_PUBLIC_POSITION_MANAGER_ADDRESS || ''
-        : DEPOSIT_POOL_ADDRESS
+  useEffect(() => {
     if (provider) {
-      if (process.env.NEXT_PUBLIC_ENVIRONMENT === 'local') {
-        if (accountInfo?.address) {
-          setPosManager(new ethers.Contract(address, PosManager.abi, provider.getSigner(accountInfo.address)))
-        } else {
-          setPosManager(new ethers.Contract(address, PosManager.abi, provider))
-        }
-      } else {
-        if (accountInfo?.address) {
-          setDepPool(new ethers.Contract(address, DepPool.abi, provider.getSigner(accountInfo.address)))
-        } else {
-          setDepPool(new ethers.Contract(address, DepPool.abi, provider))
-        }
+      if (!posManager && POSITION_MANAGER_ADDRESS) {
+        setPosManager(
+          new ethers.Contract(
+            POSITION_MANAGER_ADDRESS,
+            PosManager.abi,
+            accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+          )
+        )
       }
-    } else {
-      notifyError('Please connect wallet')
-    }
 
-    if (!accountInfo || !accountInfo.address) {
-      notifyError('Wallet not connected.')
-      return
-    }
-    if (accountInfo && accountInfo?.address) {
-      if (posManager === null) {
-        return
-      }
-      approveWithdraw(posManager, posManager.address)
-        .then(() => {
-          setEnableRemove(true)
-        })
-        .catch((err: any) => {
-          setEnableRemove(false)
-          notifyError(err)
-        })
-    } else {
-      if (depPool === null) {
-        return
+      if (!gammaPoolContract && GAMMA_POOL_ADDRESS) {
+        setGammaPoolContract(
+          new ethers.Contract(
+            GAMMA_POOL_ADDRESS,
+            GammaPool.abi,
+            accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+          )
+        )
       }
-      approveWithdraw(depPool, depPool.address)
-        .then(() => {
-          setEnableRemove(true)
-        })
-        .catch((err: any) => {
-          setEnableRemove(false)
-          notifyError(err)
-        })
     }
+  }, [provider])
+
+  async function approveTransaction() {
+    let loading = notifyLoading('Waiting for block confirmation')
+    approveWithdraw()
+      .then(() => {
+        setEnableRemove(true)
+        notifyDismiss(loading)
+      })
+      .catch((err) => {
+        notifyDismiss(loading)
+        setEnableRemove(false)
+        notifyError(err.toString())
+      })
   }
 
   async function withdrawLiquidity(balance: number) {
+    if (balance === 0) {
+      throw new Error('Please select a valid amount to withdraw')
```

Comment:
Is this for LP tokens withdrawal? Is there a case for reserve tokens withdrawal?

Reply:
What is LP tokens? From where we can select it?

Reply:
The specs for withdraw should require a switch to choose from withdraw LP tokens or reserve tokens. If you haven't started on the withdraw handler work, then we can do it in that task separately.